### PR TITLE
Add option to set custom reporter module from CLI

### PR DESCRIPTION
**Pull request by wojciechczerniak**
Automatic tool like IDE plugin or CI server can depend on custom reporter. Right now user has to configure it in `.stylintrc` file. This adds configuration step or place for an error.

CLI option has the highest priority. Even if user has different reporter configured in his `.stylintrc` file the CLI will ignore it. This way he can use stylish reporter for his daily work and plugin/ci server will use whatever it needs anyway.

### CODE REVIEW

File: bin/stylint [hunk 1]
```diff
@@ -22,6 +22,11 @@ var options = yargs
 		describe: 'Run all tests, regardless of config',
 		type: 'boolean'
 	} )
+	.option( 'reporter', {
+		alias: 'r',
+		describe: 'Custom reporter npm module name',
+		type: 'string'
```
Comment (SimenB) [inline bin/stylint]: you can add `requiresArg: true`

**Comment by SimenB** [comment]
Great idea! Maybe add a test (even though they're currently red)

**Comment by wojciechczerniak** [comment]
Test will have to mock node require() function. 

I was looking into tests and for some reason init tests are commented out. Maybe we should take a look at those tests in another PR?

File: bin/stylint (hunk 1 above)
Comment (wojciechczerniak) [inline bin/stylint]: :+1:

**Comment by SimenB** [comment]
How about just calling `init` with both options and `stylintrc` and see which is set? 

And yeah, should probably fix the tests

**Comment by wojciechczerniak** [comment]
If either `--reporter` or `.stylintrc` has custom reporter set, `init` will call `require()`. Test will fail with exception `module not found` or pass if you have a reporter installed. Adding reporter as a dependency just seems wrong.

BTW. `try-catch` those requires would allow fallback (good) to default reporter or exit with nice, informative error (better) why stylint can't output expected format, instead of exception.

**Comment by SimenB** [comment]
I agree that we should perhaps give a nicer message to the user than the current `module not found`, yeah. Different PR though.

This LGTM then

**Pull request merged**